glusterd2/commands/volumes/volume-create-txn.go

```diff
@@ -144,16 +147,23 @@ func newVolinfo(req *api.VolCreateReq) (*volume.Volinfo, error) {
 		volinfo.Transport = "tcp"
 	}
 
-	if err := populateSubvols(volinfo, req); err != nil {
-		return nil, err
-	}
-
 	if req.Metadata != nil {
 		volinfo.Metadata = req.Metadata
 	} else {
 		volinfo.Metadata = make(map[string]string)
 	}
 
+	if req.Size > 0 {
+		//AutoProvisioned volume
+		volinfo.Metadata[utils.ProvisionKey] = string(utils.AutoProvisioned)
```

Comment:
Any reason why you chose `Metadata` over adding a field to `Volinfo` ?

Reply:
That was based on previous review comments